pkg/reconciler/knativeserving/ingress/ingress_test.go

```diff
@@ -21,52 +21,51 @@ import (
 	"os"
 	"testing"
 
+	mf "github.com/manifestival/manifestival"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes/scheme"
-
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-
-	mf "github.com/manifestival/manifestival"
+	"k8s.io/client-go/kubernetes/scheme"
 	servingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
 	"knative.dev/operator/pkg/reconciler/common"
 	util "knative.dev/operator/pkg/reconciler/common/testing"
 )
 
-const numberIngressResource = 27
-
 func TestGetIngress(t *testing.T) {
 	os.Setenv(common.KoEnvKey, "testdata/kodata")
 	defer os.Unsetenv(common.KoEnvKey)
 
 	tests := []struct {
-		name                 string
-		targetVersion        string
-		expected             bool
-		expectedResourcesNum int
+		name                string
+		version             string
+		expectedNoErr       bool
+		expectedIngressPath string
 	}{{
-		name:                 "Available ingresses",
-		targetVersion:        "0.18",
-		expected:             true,
-		expectedResourcesNum: numberIngressResource,
+		name:                "Available ingresses",
+		version:             "0.18.1",
+		expectedNoErr:       true,
+		expectedIngressPath: os.Getenv(common.KoEnvKey) + "/ingress/0.18",
 	}, {
-		name:                 "Unavailable ingresses",
-		targetVersion:        "0.16",
-		expected:             false,
-		expectedResourcesNum: 0,
+		name:          "Unavailable ingresses",
+		version:       "0.16.1",
+		expectedNoErr: false,
 	}, {
-		name:                 "Missing version",
-		targetVersion:        "",
-		expected:             true,
-		expectedResourcesNum: 0,
+		name:                "Missing version",
+		version:             "",
+		expectedNoErr:       true,
+		expectedIngressPath: "",
 	}}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			manifest, _ := mf.ManifestFrom(mf.Slice{})
-			err := getIngress(tt.targetVersion, &manifest)
-			util.AssertEqual(t, err == nil, tt.expected)
-			util.AssertEqual(t, len(manifest.Resources()), tt.expectedResourcesNum)
+			err := getIngress(tt.version, &manifest)
+			util.AssertEqual(t, err == nil, tt.expectedNoErr)
```

Comment:
Can we check the `os.IsNotExist(err)` accurately instead of `err == nil`?

Reply:
I will change the `expectError` into a real error, instead of the bool.
If it is nil, check the equality. if it is an error, check the string of the error.